# Laboratório 01 — Conhecendo os Dados

**Disciplina:** Aprendizagem Computacional  
**Unidade I:** Fundamentos e Preparação dos Dados  
**Base:** Iris  
**Tempo estimado:** 40–50 minutos

## Objetivos
- carregar uma base de dados;
- reconhecer linhas, colunas e tipos de atributos;
- identificar valores ausentes;
- produzir estatísticas descritivas;
- registrar interpretações iniciais.

## 1. Preparação do ambiente

In [1]:
import pandas as pd
import numpy as np
from sklearn.datasets import load_iris

pd.set_option("display.max_columns", None)

## 2. Carregamento da base

In [2]:
iris = load_iris(as_frame=True)
df = iris.frame.copy()
df["species"] = df["target"].map(dict(enumerate(iris.target_names)))
df.head()

,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),target,species
0,5.1,3.5,1.4,0.2,0,setosa
1,4.9,3.0,1.4,0.2,0,setosa
2,4.7,3.2,1.3,0.2,0,setosa
3,4.6,3.1,1.5,0.2,0,setosa
4,5.0,3.6,1.4,0.2,0,setosa


## 3. Estrutura dos dados

In [3]:
print(f"Linhas: {df.shape[0]}")
print(f"Colunas: {df.shape[1]}")
display(df.head())
display(df.tail())

Linhas: 150
Colunas: 6


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),target,species
0,5.1,3.5,1.4,0.2,0,setosa
1,4.9,3.0,1.4,0.2,0,setosa
2,4.7,3.2,1.3,0.2,0,setosa
3,4.6,3.1,1.5,0.2,0,setosa
4,5.0,3.6,1.4,0.2,0,setosa


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),target,species
145,6.7,3.0,5.2,2.3,2,virginica
146,6.3,2.5,5.0,1.9,2,virginica
147,6.5,3.0,5.2,2.0,2,virginica
148,6.2,3.4,5.4,2.3,2,virginica
149,5.9,3.0,5.1,1.8,2,virginica


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   target             150 non-null    int64  
 5   species            150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


### Atividade 1
Responda em uma célula de texto:
1. Quantos registros existem?
2. Quantos atributos numéricos existem?
3. Qual coluna representa a classe?

**Resposta:**

1. Quantos registros existem? 150
2. Quantos atributos numéricos existem? 5
3. Qual coluna representa a classe? target

## 4. Qualidade inicial

In [5]:
resumo_qualidade = pd.DataFrame({
    "tipo": df.dtypes.astype(str),
    "ausentes": df.isna().sum(),
    "unicos": df.nunique()
})
resumo_qualidade

,tipo,ausentes,unicos
sepal length (cm),float64,0,35
sepal width (cm),float64,0,23
petal length (cm),float64,0,43
petal width (cm),float64,0,22
target,int64,0,3
species,object,0,3


## 5. Estatística descritiva

In [6]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
sepal length (cm),150.0,5.843333,0.828066,4.3,5.1,5.80,6.4,7.9
sepal width (cm),150.0,3.057333,0.435866,2.0,2.8,3.00,3.3,4.4
petal length (cm),150.0,3.758000,1.765298,1.0,1.6,4.35,5.1,6.9
petal width (cm),150.0,1.199333,0.762238,0.1,0.3,1.30,1.8,2.5
target,150.0,1.000000,0.819232,0.0,0.0,1.00,2.0,2.0


In [ ]:
df.groupby("species").mean(numeric_only=True)

### Atividade 2
Registre pelo menos **três observações** sobre a base. Considere diferenças entre espécies, dispersão e possíveis valores incomuns.

**Observações:**

1.
2.
3.

## 6. Desafio de extensão

In [7]:
# Escolha uma coluna numérica e calcule manualmente:
# média, mediana, mínimo, máximo e amplitude.

coluna = "sepal length (cm)"

media = df[coluna].mean()
mediana = df[coluna].median()
minimo = df[coluna].min()
maximo = df[coluna].max()
amplitude = maximo - minimo

media, mediana, minimo, maximo, amplitude

(np.float64(5.843333333333334), 5.8, 4.3, 7.9, 3.6000000000000005)

## Entrega
Salve o notebook com o padrão:

`LAB01_NomeSobrenome.ipynb`

Mantenha as células executadas e complete todas as respostas.